packages/core/src/queries/user.ts

```diff
@@ -36,3 +38,29 @@ export const hasUserWithId = async (id: string) =>
 `);
 
 export const insertUser = buildInsertInto<UserDBEntry>(pool, Users, { returning: true });
+
+export const findAllUsers = async () =>
+  pool.many<UserDBEntry>(sql`
+    select ${sql.join(Object.values(fields), sql`, `)}
+    from ${table}
+  `);
+
+const updateUser = buildUpdateWhere<UserDBEntry>(pool, Users, true);
+
+export const updateUserById = async (id: string, set: Partial<OmitAutoSetFields<UserDBEntry>>) =>
+  updateUser({ set, where: { id } });
+
+export const deleteUserById = async (id: string) => {
+  const { rowCount } = await pool.query(sql`
```

Comment:
do we need to catch the exceptions for this query?

Reply:
simalar to [LOG-310](https://linear.app/silverhand/issue/LOG-310/handle-core-api-404-exception), maybe all database related exeptions can be handled by one middleware